File: ServerApp/ServerApp/GameState/ChestMimic.cs
namespace ServerApp.GameState
{
    public class ChestMimic : IInteractiveObject
    {
        public string Name {get;}
        public int damage {get; set;}
        public bool discovered {get;set;}

        public ChestMimic() {
            Name = "ChestMimic";
            damage = 10;
            discovered = false;
        }
        
        public void interact(IGameState gameState)
        {
            gameState.Player.TakeDamage(damage);
            discovered = true;
        }
    }
}

File: clientApp/src/GameState/ChestMimic.ts
import { IInteractiveObject } from './IInteractiveObject';

export class ChestMimic implements IInteractiveObject {
    public name: string;
    public discovered: boolean;

    constructor(name: string, discovered: boolean) {
        this.name  = name;
        this.discovered = discovered;
    }
}
